### PR TITLE
[RN] Add avatar to remote video menu

### PR DESCRIPTION
**Pull request by zbettenbuk**
@saghul This is a follow-up PR for the remote video menu, based on your feedback. Please feel free to post-mortem review the original PR and add all your comments here for addressing.

### CODE REVIEW

**Comment by zbettenbuk** [comment]
WIP, do not review yet

File: react/features/remote-video-menu/components/native/RemoteVideoMenu.js [hunk 1]
```diff
@@ -99,10 +116,12 @@ class RemoteVideoMenu extends Component<Props> {
  *  }}
  */
 function _mapStateToProps(state, ownProps) {
-    const { id } = ownProps.participant;
+    const { participant } = ownProps;
 
     return {
-        _participantDisplayName: getParticipantDisplayName(state, id)
+        _avatarURL: getAvatarURL(participant),
+        _participantDisplayName: getParticipantDisplayName(
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/remote-video-menu/components/native/RemoteVideoMenu.js]: This is a great question. The local reason is that the ```getParticipantDisplayName``` function retrieves the participant from state using the ID, while the ```getAvatarURL``` uses a full participant object, therefore it doesn't need a state reference. I prefer passing full objects instead of just IDs, so I'd prefer the ```getAvatarURL``` way of doing it, but the ```getParticipantDisplayName``` function is used in many places of the application and I'm not sure if the nearby participant object is always a full object, not just a constructed dummy one with an ID, so I'm hesitant to change the signature.

[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/remote-video-menu/components/native/RemoteVideoMenu.js]: Never mind, I decided to change the signature anyhow, it's here in a separate commit.

**Comment by zbettenbuk** [comment]
@saghul finished, this is now ready for a new review.

Added the post-PR improvements of the remote video menu as well, and the consolidation of the function you mentioned above.

Thanks in advance!

File: react/features/base/react/components/native/Container.js [hunk 2]
```diff
@@ -50,19 +59,24 @@ export default class Container<P: Props> extends AbstractContainer<P> {
...
-                        : TouchableWithoutFeedback,
-                    {
-                        accessibilityLabel,
-                        accessible,
-                        onLongPress,
-                        onPress: onClick,
-                        ...touchFeedback && { underlayColor }
-                    },
-                    element);
+                = touchFeedback
```
Comment (saghul) [inline react/features/base/react/components/native/Container.js]: Why is this change necessary? I find it harder to read than the previous implementation.

File: react/features/filmstrip/components/native/Thumbnail.js [hunk 3]
```diff
@@ -169,43 +168,55 @@ class Thumbnail extends Component<Props> {
...
+        /**
+         * Handles click/tap event on the thumbnail.
+         *
+         * @protected
+         * @returns {void}
+         */
+        _onClick() {
+            const { participant } = ownProps;
+
+            // TODO The following currently ignores
```
Comment (saghul) [inline react/features/filmstrip/components/native/Thumbnail.js]: Let's drop this since we are here. There is no filmStripOnly mode on mobile.

File: react/features/remote-video-menu/components/AbstractKickButton.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * The function to be used to translate i18n labels.
+     */
+    t: Function
+};
+
+/**
+ * An abstract remote video menu button which kicks the remote participant.
+ */
+export default class AbstractKickButton extends AbstractButton<Props, *> {
```
Comment (saghul) [inline react/features/remote-video-menu/components/AbstractKickButton.js]: Can we make a separate commit for this? This change was not part of my suggested improvements, it's a new feature if you will, and it's very self-contained.

File: react/features/remote-video-menu/components/AbstractKickButton.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    accessibilityLabel = 'toolbar.accessibilityLabel.audioRoute';
```
Comment (saghul) [inline react/features/remote-video-menu/components/AbstractKickButton.js]: wrong label

File: react/features/remote-video-menu/components/AbstractKickRemoteParticipantDialog.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    /**
+     * Function to translate i18n labels.
+     */
+    t: Function
+};
+
+/**
+ * Abstract dialog to confirm a remote participant kick action.
+ */
+export default class AbstractKickRemoteParticipantDialog
```
Comment (saghul) [inline react/features/remote-video-menu/components/AbstractKickRemoteParticipantDialog.js]: Can we call this `KickParticipantDialog` instead? It's pretty obvious we are dealing with remote participants.

File: react/features/remote-video-menu/components/AbstractMuteButton.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+     * The function to be used to translate i18n labels.
+     */
+    t: Function
+};
+
+/**
+ * An abstract remote video menu button which mutes the remote participant.
+ */
+export default class AbstractMuteButton extends AbstractButton<Props, *> {
+    accessibilityLabel = 'toolbar.accessibilityLabel.audioRoute';
```
Comment (saghul) [inline react/features/remote-video-menu/components/AbstractMuteButton.js]: wrong label

File: react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
+/* @flow */
```
Comment (saghul) [inline react/features/remote-video-menu/components/AbstractMuteRemoteParticipantDialog.js]: nit: use `// @flow`

File: react/features/remote-video-menu/components/web/MuteButton.js [hunk 9]
```diff
@@ -87,34 +56,12 @@ class MuteButton extends Component<Props> {
                 displayClass = { muteConfig.muteClassName }
                 iconClass = 'icon-mic-disabled'
                 id = { `mutelink_${participantID}` }
-                onClick = { this._onClick } />
+                // eslint-disable-next-line react/jsx-handler-names
```
Comment (saghul) [inline react/features/remote-video-menu/components/web/MuteButton.js]: can't we avoid this by calling the handler `onSomethingClick` ?

File: react/features/remote-video-menu/components/native/styles.js [hunk 10]
```diff
@@ -15,6 +15,7 @@ export default createStyleSheet({
         color: ColorPalette.lightGrey,
         flexShrink: 1,
         fontSize: 16,
+        marginLeft: 16,
```
Comment (saghul) [inline react/features/remote-video-menu/components/native/styles.js]: This value also comes from MD, it should be a constant.

File: react/features/base/participants/functions.js [hunk 11]
```diff
@@ -145,17 +145,10 @@ export function getParticipantCountWithFake(stateful: Object | Function) {
...
- * @private
+ * @param {Object} participant - Participant object.
  * @returns {string}
  */
-export function getParticipantDisplayName(
-        stateful: Object | Function,
-        id: string) {
-    const participant = getParticipantById(stateful, id);
-
+export function getParticipantDisplayName(participant: Object) {
```
Comment (saghul) [inline react/features/base/participants/functions.js]: I don't think this is good going forward. `interfaceConfig` will eventually be part of the state, so how would we fetch stuff from it, if we don't have access to it?

File: react/features/base/react/components/native/Container.js (hunk 2 above)
Comment (zbettenbuk) [reply react/features/base/react/components/native/Container.js]: Because we were feeding an invalid prop to the component when ```touchFeedback``` was false

File: react/features/remote-video-menu/components/AbstractKickButton.js (hunk 4 above)
Comment (zbettenbuk) [reply react/features/remote-video-menu/components/AbstractKickButton.js]: I don't think this is a new feature. This should be like this originally when I implemented it (separate abstract and platform dependant code), so I'd leave it here, if you don't mind

Comment (saghul) [reply react/features/remote-video-menu/components/AbstractKickButton.js]: Ok

File: react/features/remote-video-menu/components/AbstractKickButton.js (hunk 5 above)
Comment (zbettenbuk) [reply react/features/remote-video-menu/components/AbstractKickButton.js]: Ehm... indeed... 🤕

File: react/features/remote-video-menu/components/web/MuteButton.js (hunk 9 above)
Comment (zbettenbuk) [reply react/features/remote-video-menu/components/web/MuteButton.js]: The name ```_handleClick``` comes from the ```AbstractButton``` implementation. So I either had to do this, or encapsulate (or re-bind) it into a meaningless function that starts with ```_on``` but simply invokes ```_handleClick```. Both are kind of pointless, but I picked this solution to avoid unnecessary code. I was thinking: the entire render method will be removed anyhow when we refactor web to use the same abstraction.

File: react/features/base/participants/functions.js (hunk 11 above)
Comment (zbettenbuk) [reply react/features/base/participants/functions.js]: Well, currently we don't need the state object in those calls. Would you rather refactor these to get the participant from the state instead of using the passed value only for the sake of having the state object available there for some future use? I wouldn't, but I'm curious what you think.

File: react/features/remote-video-menu/components/AbstractKickRemoteParticipantDialog.js (hunk 6 above)
Comment (zbettenbuk) [reply react/features/remote-video-menu/components/AbstractKickRemoteParticipantDialog.js]: Well, it's a matter of consistency vs simplification :)

Consistency: The mute dialog is also named MuteRemoteParticipantDialog, so we can stay consistent with that (or rename that too. Question if that's misleading without the "remote" word or not) or...
Simplification: Rename this dialog, but then see previous point.

What do you think?

Comment (saghul) [reply react/features/remote-video-menu/components/AbstractKickRemoteParticipantDialog.js]: Let's stay consistent then, good point!

File: react/features/remote-video-menu/components/web/MuteButton.js (hunk 9 above)
Comment (saghul) [reply react/features/remote-video-menu/components/web/MuteButton.js]: Ok.

File: react/features/base/participants/functions.js (hunk 11 above)
Comment (saghul) [reply react/features/base/participants/functions.js]: I'd have a function which takes `state` and `participant`. It uses each as needed. Now we are not using state and we know this, but we will, so let's not put ourselves in a position where we need to do it again later...
